ironfist/src/cpp/scripting/scripting.h

```diff
@@ -1,6 +1,16 @@
 #ifndef SCRIPTING_H
 #define SCRIPTING_H
 
+#include <map>
+
+typedef std::map<std::string, std::pair<std::string, std::string>> luaTable;
+
+struct mapVariable {
+	std::string luaType;
+	std::string singleValue;

```

Comment:
The way you have this now, every mapVariable has space for both a singleValue and a tableValue.

More fundamentally, this type definition is telling you a mapVariable has both a singleValue and a tableValue, when it only has one.

Also, the strings for the Lua type don't really tell you what should be in that field; you have to keep doing string comparisons.

Here's how I'd design this data structure:

enum MapVarType { MAPVAR_TYPE_STRING, MAPVAR_TYPE_INT, MAPVAR_TYPE_TABLE, ... };

struct mapVariable {
  MapVarType type;
  union {
    long long intValue; // I think ints in Lua are 64-bit; you can check;
    double floatValue;
    luaTable tableValue;
   ...
  } value;
}